cmake/FindLua.cmake

```diff
@@ -0,0 +1,260 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
+# file Copyright.txt or https://cmake.org/licensing for details.
+
+#[=======================================================================[.rst:
+FindLua
+-------
+
+Locate Lua library compiled as C++ code.
+
+This module defines:
+
+``LUA_FOUND``
+  if false, do not try to link to Lua
+``LUA_LIBRARIES``
+  both lua and lualib
+``LUA_INCLUDE_DIR``
+  where to find lua.h
+``LUA_VERSION_STRING``
+  the version of Lua found
+``LUA_VERSION_MAJOR``
+  the major version of Lua
+``LUA_VERSION_MINOR``
+  the minor version of Lua
+``LUA_VERSION_PATCH``
+  the patch version of Lua
+
+Note that the expected include convention is
+
+::
+
+  #include <lua.h>
+
+and not
+
+::
+
+  #include <lua/lua.h>
+
+nor
+
+::
+
+  #include "lua.h"
+
+This is because the Lua location is not standardized and may exist in locations
+other than lua/, and Wesnoth source includes a conflicting header named lua.h.
```

Comment:
The "Wesnoth includes a header named lua.h" doesn't seem to be true anymore.

This section of the documentation doesn't say which files this convention applies to. Is it for files expecting to include (what is currently) Wesnoth's lua.h, is it for the wrapper that includes system Lua, or is it for the small .cpp test file that CMake might make to check that system Lua is present?

Reply:
Right, that's long outdated.  Good catch, thanks.  I [updated and clarified](https://github.com/wesnoth/wesnoth/compare/fdd34cb62c9..081cec1932b) the documentation.  I hope that answers your questions.